packages/react-table/src/components/Table/utils/decorators/headerCol.tsx

```diff
@@ -6,7 +6,7 @@ export const headerCol = (id = 'simple-node') => {
     const result = typeof value === 'object' ? value.title : value;
     return {
       component: 'th',
-      children: <div id={`${id}${rowIndex}`}>{result}</div>
+      children: <div id={`${id}${rowIndex}`}>{result as React.ReactNode}</div>
```

Comment:
surprised by this change to Table. is this necessary?

Reply:
@nicolethoen -- so it was a cascade of events:
tl:dr; I had to upgrade the root typescript version which was at 4.0.0 to 4.7.4 to match the versions of all the other packages including the demo app -- and typescript being what it is added more semantic checks which appeared on these files

If I didn't make the root typescript match, importing those packages to the demo app for hot replacement caused those typescript errors to appear

Reply:
@jeswanke I see. I'm just going to tag @tlabaj to double check that it doesn't have further ripple effects i'm not seeing.

Reply:
This is fine since the type of the `title` in `IFormatterValueType` is a ReactNode or string (thought string ins not needed sine ReactNode includes a string).   
```
export type IFormatterValueType = formatterValueType & {
  title?: string | React.ReactNode;
  props?: any;
};
```